src/amrex/extensions/MultiFab.py

```diff
@@ -173,6 +183,496 @@ def copy_multifab(amr, self):
+
+    Parameters
+    ----------
+    mfi : MFIter
+        The MFIter instance for the current block,
+    starts : sequence
+        The minimum indices of the global slice.
+        These can be negative.
+    stops : sequence
+        The maximum indices of the global slice.
+        These can be negative.
+    icstart : integer
+        The minimum component index of the global slice.
+        These can be negative.
+    icstops : integer
+        The maximum component index of the global slice.
+        These can be negative.
+    include_ghosts : bool, default=False
+        Whether or not ghost cells are included
+    with_internal_ghosts: bool
+        Whether the internal ghosts are included in the slices
+
+    Returns
+    -------
+    block_slices : tuple or None
+        The slices of the intersections relative to the block
+    global_slices : tuple or None
+        The slices of the intersections relative to the global array where the data from individual block will go
+    """
+    box = mfi.tilebox()
+    box_small_end = box.small_end
+    box_big_end = box.big_end
+    if include_ghosts:
+        nghosts = self.mf.n_grow_vect
+        box.grow(nghosts)
+        if with_internal_ghosts:
+            box_small_end = box.small_end
+            box_big_end = box.big_end
+        else:
+            min_box = self.box_array().minimal_box()
+            for i in range(len(nghosts)):
+                if box_small_end[i] == min_box.small_end[i]:
+                    box_small_end[i] -= nghosts[i]
+                if box_big_end[i] == min_box.big_end[i]:
+                    box_big_end[i] += nghosts[i]
+
+    boxlo = _get_indices(box.small_end, 0)
+    ilo = _get_indices(box_small_end, 0)
+    ihi = _get_indices(box_big_end, 0)
+
+    # Add 1 to the upper end to be consistent with the slicing notation
+    ihi_p1 = [i + 1 for i in ihi]
+    i1 = np.maximum(starts, ilo)
+    i2 = np.minimum(stops, ihi_p1)
+
+    if np.all(i1 < i2):
+        block_slices = []
+        global_slices = []
+        for i in range(3):
+            block_slices.append(slice(i1[i] - boxlo[i], i2[i] - boxlo[i]))
+            global_slices.append(slice(i1[i] - starts[i], i2[i] - starts[i]))
+
+        block_slices.append(slice(icstart, icstop))
+        global_slices.append(slice(0, icstop - icstart))
+
+        return tuple(block_slices), tuple(global_slices)
+    else:
+        return None, None
+
+
+def __getitem__(self, index):
+    """Returns slice of the MultiFab using global indexing.
+    The shape of the object returned depends on the number of ix, iy and iz specified, which
+    can be from none to all three. Note that the values of ix, iy and iz are
+    relative to the fortran indexing, meaning that 0 is the lower boundary
+    of the whole domain, and in fortran ordering, i.e. [ix,iy,iz].
+    This allows negative indexing, though with ghosts cells included, the first n-ghost negative
+    indices will refer to the lower guard cells.
```

Comment:
For `__getitem__` and `__setitem__`, I would add a bit more info of the form:
```suggestion
    indices will refer to the lower guard cells.

    Note: in MPI-parallel contexts, this triggers a...
```
Please see more details below on `__setitem__`: